### ISSUE TITLE
in `#[cfg(test)]` modules, attribute macros have error highlighting when they shouldn't

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#9855]
## Environment

* **IntelliJ Rust plugin version:** 0.4.185.5058-223-nightly
* **Rust toolchain version:** 1.67.0-nightly (e0098a5cc 2022-11-29) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2022.3 Ultimate Edition (IU-223.7571.182)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr

## Problem description
Actually, now the plugin shouldn't show any errors inside attribute macros (to avoid false positives). 
<img width="605" alt="image" src="https://user-images.githubusercontent.com/49211026/206002383-0d0eba17-e3de-49f7-bdf2-f5eff8112801.png">
(error on line 10 is "false" and will be fixed by macro expansion someday. Error on line 11 is correct, but should be suppressed)

Normal behaviour without `#[cfg(test)]`:
<img width="605" alt="image" src="https://user-images.githubusercontent.com/49211026/206002416-015164a7-bc15-499e-856e-9c5342c785f7.png">

## Steps to reproduce
```rust
// #[cfg(test)]
mod test {
    #[mysecondlib::attr_add_to_fn_beginning(let iii = 6;)]
    fn foo() {
        let r = iii; // <--- show type info works fine
        let w = unresolved;
        println!("{}", r);
    }
}
```

macros definition in `mysecondlib` crate:
```rust
use proc_macro::{TokenTree, Delimiter, Group, TokenStream};
#[proc_macro_attribute]
pub fn attr_add_to_fn_beginning(attr: TokenStream, item: TokenStream) -> TokenStream {
    item.into_iter().map(|tt| {
        match tt {
            TokenTree::Group(g) => {
                if g.delimiter() == Delimiter::Brace {
                    let mut dg = Group::new(
                        g.delimiter(),
                        attr.clone().into_iter().chain(g.stream().into_iter()).collect()
                    );
                    dg.set_span(g.span());
                    TokenTree::Group(dg)
                } else {
                    TokenTree::Group(g)
                }
            }
            TokenTree::Ident(..) => tt,
            TokenTree::Punct(..) | TokenTree::Literal(..) => tt,
        }
    }).collect()
}
```
